### ISSUE TITLE
Make the GlobalSetOption hook part of a group

### ISSUE THREAD

**Issue by EmmChriss** [TeddyDD/kakoune-lf#3]
This isn't strictly needed, but I had a very hard time getting it to work with both tmux, and x11 terminal.
`hatch-terminal` is only supposed to expose the `KAK_CLIENT`, and `KAK_SESSION` variables, the other ones in lowercase are just a side-effect of the terminal being directly spawned from the kak process, inheriting its environment. 
On the tmux side, this is different, as spawning a pane is done through the tmux command, which communicates with the server through a pipe, and drops its environment.
The `KAK_(CLIENT|SESSION)` variables however are deliberately made part of the hatched pane's environment, so they're available.
Switching the configuration is mostly possible, but the hook has to be removed, so that it can be redefined with the uppercase variants, but that is not possible without giving it a group.
#### TL;DR: Please add the hook to some group 

Edit: Also it'd nice to change the way lf is configured when the connection is made

**Comment by TeddyDD** [comment]
I wonder if overriding `lf_terminal_cmd` when you are in tmux isn't enough for this case. I can add hooks to some group since that's good idea anyway

**Comment by EmmChriss** [comment]
Well, yeah, it's possible without adding the group, but this seems like a best practice, and also makes customization easier in my opinion

**Comment by TeddyDD** [comment]
deeea9db36b96412cfbdfab545d4851a2946671f is this ok for you?

**Comment by EmmChriss** [comment]
Yeah, thanks
